src/components/shared/chart/Viewport.js

```diff
@@ -463,6 +511,94 @@ export const withChartViewport: WithChartViewport<*, *> =
         this.moveViewport(offsetX, offsetY);
       };
 
+      _keyDownListener = (
+        event: { nativeEvent: KeyboardEvent } & SyntheticKeyboardEvent<>
+      ) => {
+        let navigationKey;
+        if (
+          !event.ctrlKey &&
+          !event.shiftKey &&
+          !event.altKey &&
+          !event.metaKey
+        ) {
+          navigationKey = BARE_KEYMAP[event.nativeEvent.code];
+        } else if (event.ctrlKey) {
+          navigationKey = CTRL_KEYMAP[event.nativeEvent.code];
+        }
+
+        if (navigationKey !== undefined) {
+          // Start requesting frames if there were no keys down before
+          // this event triggered.
+          if (this._keysDown.size === 0) {
+            requestAnimationFrame(this._keyboardNavigation);
+            this._lastKeyboardNavigationFrame = performance.now();
+          }
+
+          this._keysDown.add(navigationKey);
+          event.preventDefault();
+        }
+      };
+
+      _keyUpListener = (
+        event: { nativeEvent: KeyboardEvent } & SyntheticKeyboardEvent<>
+      ) => {
+        if (!event.ctrlKey) {
+          // The ctrl modifier might have been released here. Try to
+          // delete all keys associated with the modifier. Since the
+          // navigation is aliased with non-ctrl-modified keys also,
+          // this will affect (stop) the operation even if it was
+          // introduced without a ctrl-modified key.
+          for (const code of Object.keys(CTRL_KEYMAP)) {
```

Comment:
I had to use `Object.keys` rather than `Object.values` to keep Flow happy, so this is a workaround. Is there a better way to do this without making the code less obvious?

Reply:
See `src/utils/flow.js` and `objectEntries`, and `objectValues`. Perhaps that approach could help here.